vivisect/tests/testvivisect.py

```diff
@@ -50,30 +58,223 @@ def test_cli_search(self):
+        output = self.chgrp_vw.canvas.strval
+        self.assertIn("searching from 0x08048000 for 512 bytes\npattern not found: 71736f7274 (b'qsort')\n", output)
+        self.chgrp_vw.canvas.clearCanvas()
+        
         self.chgrp_vw.do_search("-c -r qsort -R 0x8048000:0x2000")
+        output = self.chgrp_vw.canvas.strval
+        self.assertIn("0x80488e3  71 73 6f 72 74 00 66 63 6e 74 6c 00 6d 65 6d 6d   qsort.fcntl.memm\n0x80488f3  6f 76 65 00 62 69 6e 64 74 65 78 74 64 6f 6d 61   ove.bindtextdoma\n\ndone (1 results).\n", output)
+        self.chgrp_vw.canvas.clearCanvas()
+        
 
     def test_cli_searchopcode(self):
         '''
         Test that VivCli.do_searchopcodes works
         '''
-        #TODO: make real tests with asserts
-        self.chgrp_vw.do_searchopcodes('foo')
+        # i have no idea why this is necessary (we did it at setup time)
+        # but the current renderer at this point is envi.memcanvas.renderers.ByteRend
+        self.chgrp_vw.canvas.setRenderer('viv')
+        self.chgrp_vw.do_searchopcodes('sar')
+        output = self.chgrp_vw.canvas.strval
+        self.assertIn(".text:0x08049294  d1f8             sar eax,1\n\n", output)
+        self.chgrp_vw.canvas.clearCanvas()
+
         self.chgrp_vw.do_searchopcodes('-f 0x08050200 ret')
+        output = self.chgrp_vw.canvas.strval
+        self.assertIn("matches for: 726574 ('ret')\n.text:0x0805020a  c3               ret \n\ndone (1 results).\n", output)
+        self.chgrp_vw.canvas.clearCanvas()
+
         self.chgrp_vw.do_searchopcodes('-c rol')
+        output = self.chgrp_vw.canvas.strval
+        self.assertIn("pattern not found: 726f6c ('rol')\n", output)
+        self.chgrp_vw.canvas.clearCanvas()
+
         self.chgrp_vw.do_searchopcodes('-o rol')
+        output = self.chgrp_vw.canvas.strval
+        self.assertIn("pattern not found: 726f6c ('rol')\n", output)
+        self.chgrp_vw.canvas.clearCanvas()
+
         self.chgrp_vw.do_searchopcodes('-t rol')
+        output = self.chgrp_vw.canvas.strval
+        self.assertIn(".text:0x08050249  66d3c0           rol ax,cl", output)
+        self.chgrp_vw.canvas.clearCanvas()
+
         self.chgrp_vw.do_searchopcodes('-M red rol')
+        output = self.chgrp_vw.canvas.strval
+        self.assertIn(".text:0x08050268  d2c0             rol al,cl", output)
+        self.chgrp_vw.canvas.clearCanvas()
+
         self.chgrp_vw.do_searchopcodes('-f 0x08050200 -R r.t')
+        output = self.chgrp_vw.canvas.strval
+        self.assertIn("matches for: 722e74 ('r.t')\n.text:0x0805020a  c3               ret \n\n", output)
+        self.chgrp_vw.canvas.clearCanvas()
+
+    def test_cli_argtrack(self):
+        self.chgrp_vw.do_argtrack('')
+        output = self.chgrp_vw.canvas.strval
+        self.assertIn('Usage: argtrack <func_addr_expr> <arg_idx>', output)
+        self.chgrp_vw.canvas.clearCanvas()
+
+        ## FIXME: make tests when this works
```

Comment:
?

Reply:
argtrack is the one we've leaving broken for the moment.  too complicated and misunderstood to bring it into the 2020's (last time it worked was 2013)

Reply:
i've pulled this test as well as renamed `do_argtrack` to avoid it being "another broken Vivisect thing".  we know it's broken, we aren't able to fix it right now, so just keep people from trying it and having it broken.  it's an awesome functionality if and when we can revamp it to use apiCall and VivTaints.

Reply:
to clarify:  `argtrack` will no longer show up in the CLI.